docker/Dockerfile

```diff
@@ -84,55 +127,47 @@ COPY development/nautobot_config.py /opt/nautobot/nautobot_config.py
 ENTRYPOINT ["/docker-entrypoint.sh"]
 CMD ["nautobot-server", "runserver", "0.0.0.0:8080", "--insecure"]
 
-# Copy nautobot source and poetry info to container to reduce cache invalidations
-COPY poetry.lock pyproject.toml README.md /source/
-COPY nautobot /source/nautobot
-
-# Build the whl for use in the final container and install for the dev container
-RUN poetry build && \
-    poetry install --no-ansi --extras all
+# Install development-specific dependencies of Nautobot as well as Nautobot itself in editable mode
+RUN poetry install --no-ansi --extras all
 
-################################ Stage: cleaninstall
-# Build an image with required dependencies to pull the installation from
-FROM base as cleaninstall
+################################ Stage: install-nautobot (intermediate build target)
 
-# hadolint ignore=DL3008
-RUN apt-get update && apt-get install --no-install-recommends -y build-essential libssl-dev libxmlsec1-dev libxmlsec1-openssl pkg-config libldap-dev libsasl2-dev libmariadb-dev
+FROM install-dependencies as install-nautobot
 
-# pyuwsgi doesn't support ssl so we build it from source
-# https://github.com/nautobot/nautobot/issues/193
-RUN pip install --no-cache-dir --no-binary=pyuwsgi pyuwsgi
+# hadolint ignore=DL3013,SC2102
+RUN for whl in /source/dist/nautobot*.whl; do pip install --no-deps --no-cache-dir ${whl}[all]; done
 
-COPY --from=dev /source/dist/*.whl /tmp
+# Generate self signed ssl certificates for later use
+RUN openssl req -new -newkey rsa:4096 -days 365 -nodes -x509 -subj \
+    '/C=US/ST=NY/L=NYC/O=Nautobot/CN=nautobot.local' \
+    -keyout /opt/nautobot/nautobot.key -out /opt/nautobot/nautobot.crt
 
-# hadolint ignore=DL3013,SC2102
-RUN for whl in /tmp/nautobot*.whl; do pip install --no-cache-dir ${whl}[all]; done
+################################ Stage: final (production-ready image)
 
-################################ Stage: final
-# This image will be the production ready image
 FROM base as final
 ARG PYTHON_VER
 
 # Make sure we don't run as a root user
 RUN useradd --system --shell /bin/bash --create-home --home-dir /opt/nautobot nautobot
 
-# Copy from base the required python libraries and binaries
-COPY --from=cleaninstall /usr/local/lib/python${PYTHON_VER}/site-packages /usr/local/lib/python${PYTHON_VER}/site-packages
-COPY --from=cleaninstall /usr/local/bin /usr/local/bin
+# Copy from install-nautobot the required python libraries and binaries
```

Comment:
I think poetry ends up in the final image.  I'm not sure this is desirable.

Reply:
The final image already includes `pip` and `wheel` as well - what's the concern over including `poetry`? Appreciate your thoughts on this!

Reply:
Actually I checked and because `poetry` is installed while we're still running as `root`, it's installed to `/root/.local/` rather than `/usr/local`, which is a directory the final image's `nautobot` user doesn't have access to. So I think we're OK here, though it might not be a bad optimization to `rm -rf /root/.local/` as part of the final build?

Reply:
nah I think you are good without the /root/.local...